### ISSUE TITLE
Failed to send transaction

### ISSUE THREAD

**Issue by itsnyx** [WalletConnect/web3modal-react-native#28]
Hello , everyone.

Im testing latest example published to the github as of today and i can connect my wallet with no problem.
but when i use the all methods works except eth_sendTransaction !

can you please fix this issue.!?
![image](https://github.com/WalletConnect/web3modal-react-native/assets/74738973/bd538a5b-af95-4e26-8b2a-b1a0aad6432e)

and i have enough funds to initiate transaction but it gets suck here: 
![image](https://github.com/WalletConnect/web3modal-react-native/assets/74738973/3030c891-871b-4418-9acb-85c20bcf2fbc)

**Comment by arslanrazzaq** [comment]
@itsnyx  I tried this library in react native cli project, It's working fine if i didn't use the navigation. But throwing initialization error with the navigation.

**Comment by ignaciosantise** [comment]
@arslanrazzaq feel free to open a new issue with all the related info, screenshots and and example repo to reproduce the bug.

**Comment by ignaciosantise** [comment]
@itsnyx which wallet are you using?

**Comment by itsnyx** [comment]
@ignaciosantise i only tested wallet connect both via qr code and also from installed wallets method.

other functions such as signing message or transaction works fine

**Comment by ignaciosantise** [comment]
@itsnyx  can you test with another wallet? Trust Wallet has some issues that'll be solved in their next release.

Context: https://github.com/WalletConnect/web3modal-react-native/issues/22

**Comment by itsnyx** [comment]
@ignaciosantise i dont know this information helps or not but before using web3modal i was using this libraries directly inside app:

```
    "@walletconnect/react-native-compat": "^2.4.0",
    "@walletconnect/universal-provider": "^2.4.0",
```

and i did dependancy update to the following: 

```
    "@walletconnect/react-native-compat": "^2.5.1",
    "@walletconnect/universal-provider": "^2.5.1",
```

and after this update i could not do any blockchain actions ! 
it would not show up inside wallet apps!
and after this i downgraded to the old versions and started working 
if you are using these libraries inside web3 modal library , maybe this is the issue !

**Comment by itsnyx** [comment]
@ignaciosantise  i tested with safe pal and send transaction method actually did work with no probelm ! 
so the issue is from trust wallet side

**Comment by ignaciosantise** [comment]
@itsnyx yes, lets wait until their next release. Hope it is solved. As it's not an issue with this SDK, im closing this and i'll post updates in #22
